staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -45,6 +45,20 @@ func NewSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer runtim
 	}
 }
 
+// NewStrictSerializer is the same as NewSerializer except that it creates a strict JSON serializer
+// that can also return errors of type strictDecoderError.
+// Note that this serializer is not as performant as the non-strict variant, and should not be used in fast paths.
+func NewStrictSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer runtime.ObjectTyper, pretty bool) *Serializer {
```

Comment:
I think pretty and strict should both be in a single options struct, and we should not make N NewStrictXSerializer functions.

Reply:
that means change `func NewSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer runtime.ObjectTyper, pretty bool) *Serializer` like `func NewSerializer(meta MetaFactory, creater runtime.ObjectCreater, typer runtime.ObjectTyper, pretty bool, strict bool) *Serializer` ?  i think we should change all reference about `NewSerializer`, a lot

Reply:
We could make `NewSerializer()` and `NewYAMLSerializer()` wrap `NewSerializerWithOptions()` and add deprecation notices.
`NewSerializerWithOptions()` would then be the standard constructor to use and would support all combos of pretty, yaml, and strict.

I looked at all the places we call `NewSerializer()` and `NewYAMLSerializer()` and I think it's reasonable to do a cleanup -- it could be the same patch or a follow-up.

I do find it dissonant that we support YAML in the JSON pkg.

Thoughts? :)